File: 250424/상품 코드/product-code.java
import java.util.Scanner;

class ProductCode {
    String name;
    int code;

    public ProductCode() {
        this.name = "codetree";
        this.code = 50;
    }

    public ProductCode(String product, int productCode) {
        this.name = product;
        this.code = productCode;
    }
}

public class Main {

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        ProductCode productCode = new ProductCode();
        System.out.printf("product %d is %s\n", productCode.code, productCode.name);
        
        String name = sc.next();
        int code = sc.nextInt();

        ProductCode productCode2 = new ProductCode(name, code);
        System.out.printf("product %d is %s\n", productCode2.code, productCode2.name);
        sc.close();
    }
}

File: 250424/상품 코드/product-code.cpp
#include <iostream>
#include <string>
using namespace std;

class Product {
    public:
        string id;
        int code;
        Product(string id, int code) {
            this->id = id;
            this->code = code;
        }
};

int main()
{
    ios::sync_with_stdio(false);
    cin.tie(0);
    cout.tie(0);

    Product product1 = Product("codetree", 50);
    string id;
    int code;
    cin >> id >> code;

    Product product2 = Product(id, code);
    cout << "product " << product1.code << " is " << product1.id << "\n";
    cout << "product " << product2.code << " is " << product2.id << "\n";
    return 0;
}